cdap-ui/app/cdap/components/RouteToNamespace/index.js

```diff
@@ -46,23 +46,31 @@ export default class RouteToNamespace extends Component {
     if (!list || list.length === 0) { return; }
 
     /**
-     * 1. Check if localStorage has a 'DefaultNamespace' set by the user, if not,
-     * 2. Check if there is a 'default' namespace from backend, if not,
-     * 3. Take first one from the list of namespaces from backend.
+     * 1. Check if localStorage has an already selected namespace set by the user, if not,
+     * 2. Check if localStorage has a 'DefaultNamespace' set by the user, if not,
```

Comment:
NS holds the last accessed namespace and defaultNamespace indicates the namespace that the user has set as default. Today we use localStorage to store the default but when we start using the user store we would store it in there and if the user opens the UI in a new environment where NS wouldn't be there in the localstorage we would go to the default namespace.

The correct fix of checking valid namespace before navigating to it will come in when we start using user store though.